components/clarinet-deployments/src/requirements.rs

```diff
@@ -30,8 +45,24 @@ pub async fn retrieve_contract(
     );
 
     let response = fetch_contract(request_url).await?;
-    let code = response.source.to_string();
-    contract_location.write_content(code.as_bytes())?;
+    let code = response.source;
+
+    let _ = match file_accessor {
+        None => contract_location.write_content(code.as_bytes()),
+        Some(file_accessor) => {
+            match file_accessor
+                .write_file(
+                    manifest_location,
+                    contract_location.to_string(),
+                    code.as_bytes(),
+                )
+                .await
+            {
+                Ok(_) => Ok(()),
+                Err(err) => Err(err),
```

Comment:
Not necessarily, what do you think would be better?

Reply:
I think it would make sense to allow the caller to handle the error instead of just ignoring it. This function already returns a Result, so it should be simple.